code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/CredentialUniqueFilter.java

```diff
@@ -20,13 +22,16 @@
     SchemaFactory schemaFactory;
     @Inject
     PasswordDao passwordDao;
+    @Inject
+    JsonMapper jsonMapper;
 
     @Override
     public Object create(String type, ApiRequest request, ResourceManager next) {
         Credential credential = request.proxyRequestObject(Credential.class);
         if (StringUtils.isBlank(credential.getKind())) {
             credential.setKind(request.getType());
         }
+        LocalAuthPasswordValidator.validatePassword(credential.getSecretValue(), jsonMapper);
```

Comment:
@ibuildthecloud yes sorry I was adding the call to TransformationService.
So along with calling the validatePassword only if credential is of kind password, even the transformationService will be called only if credential is password right?